src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -101,6 +101,34 @@ protected virtual SourceText CreateText(Stream stream, Workspace workspace)
             return textAndVersion;
         }
 
+        internal override TextAndVersion LoadTextAndVersion(Workspace workspace, DocumentId documentId, CancellationToken cancellationToken)
+        {
+            DateTime prevLastWriteTime = FileUtilities.GetFileTimeStamp(_path);
```

Comment:
1. we probably should inject this somehow.  That way we can actually write tests with Mocks so that we can validate behavior with FS failures.